packages/cli/utils/version.ts

```diff
@@ -25,37 +25,41 @@ export async function current() {
 }
 
 async function getLatestVersionFromNpm(packageName: string): Promise<string> {
-  try {
-    const response = await fetch(`https://registry.npmjs.org/${packageName}`, {
-      signal: AbortSignal.timeout(5000),
-    });
+  // await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
Not sure why this happens. If we catch failed promises, even if now awaited, should not fail. What exactly is happening?

I'm not against removing the try but need to understand why rejections aren't processed.

Reply:
we did catch it, but then threw again inside the catch. That second one wasn't being catch'ed.